feat: Link to OGP toolset

tohhengxing opened the pull request:
## Problem
New navbar link on our landing page 'Gov Apps' with dropdown menu to our other apps (use our component currently on admin user link)

Closes [#2816]

## Solution
add dropdown reusable component on landing page

**Breaking Changes** 
<!-- Does this PR contain any backward incompatible changes? If so, what are they and should there be special considerations for release? -->
- [ ] Yes - this PR contains breaking changes
    - Details ...
- [x] No - this PR is backwards compatible  

**Features**:

reusable component, can add more links if you need to.

## Before & After Screenshots

**AFTER**:


https://user-images.githubusercontent.com/102740235/177200059-69e08d5d-e37e-4c6b-b364-d7f82d15b98a.mp4


## Tests
no unit tests

tohhengxing committed:
Squashed commit of the following:

commit 176b4e9b797661dd3ac0b7d8e13116cfa8f718d9
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Tue Jul 5 00:01:44 2022 +0800

    refactor css and clean up code

commit 0f0131edf30d627f16901ff41452013a9df315cc
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Mon Jul 4 22:51:59 2022 +0800

    media query feature

commit 672ef10edbef8fb84dc37fb41d596f02be4fefe9
Merge: 24d1f8ca 7d094c20
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Mon Jul 4 22:36:02 2022 +0800

    Merge branch 'develop' into ogp-links-feature

commit 24d1f8ca6204d83f71152e87deff4d69a6e0ec12
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Fri Jul 1 22:09:12 2022 +0800

    improve css

    modified css so that button behaves more accurately

commit 6c8d92cef3188fc1779d5b10a88400231f59cb51
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Fri Jul 1 21:12:32 2022 +0800

    finish component for non media view

    1. left media view for mobile to implement

commit 12bed53d583643e900a9c0ee2eee4f8a4dae8b89
Merge: 1506e602 677fd2de
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Fri Jul 1 02:19:22 2022 +0800

    Merge branch 'develop' into ogp-links-feature

commit 1506e602f18ed955efa30b1967a5246690221779
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Fri Jul 1 02:18:15 2022 +0800

    modified css

commit 2c3959aeebf08f2bb6ecb3c040c2cd48358d0600
Author: tohhengxing <tohhengxing@protonmail.com>
Date:   Thu Jun 30 02:04:11 2022 +0800

    dropdown ogp link component

    1. built component
    2. but css is not done yet, need figure out how to fix

Update landing.client.view.html

Update avatar-dropdown.css

tohhengxing committed:
linting

1. fix lint issues

timotheeg commented on src/public/modules/core/components/toolset-dropdown.client.component.js:
```diff
@@ -0,0 +1,34 @@
+  vm.isDropdownOpen = false
+  vm.labelAndLink = [
+    { label: 'Short Links', link: 'https://go.gov.sg/' },
+    { label: 'Mass Messaging', link: 'https://go.gov.sg/postmangovsg' },
+    { label: 'Eligibility Checker', link: 'https://go.gov.sg/checkfirstgovsg' },
+    { label: 'Voucher Distribution', link: 'https://go.gov.sg/redeemgovsg' },
+    { label: 'Website Builder', link: 'https://go.gov.sg/isomergovsg' },
+    { label: 'Payments', link: 'https://go.gov.sg/pay' },
+    { label: 'Citizen Identity', link: 'https://go.gov.sg/idgovsg' },
```
Are all indirections through go.gov.sg for all the product links really needed? (e.g. `https://go.gov.sg/redeemgovsg` vs. `https://redeem.gov.sg/`)

tohhengxing replied:
Hi, I just followed the specifications laid out by issue #2816 . Has there been a consensus made on whether to use the direct link or go.gov?

mantariksh replied:
let's just follow the spec, no major difference

timotheeg commented on src/public/translations/en-SG/landing.json:
```diff
@@ -8,6 +8,9 @@
     "HREF": "@:LINKS.GITHUB"
   },
+  "GOV_APPS": {
+    "TEXT": "Gov Apps"
```
@cakesoccer @shannenho, is the heading OK with you? All the links are to OGP products, so just double checking whether "Gov Apps" works for you or if we want a slightly more specific branded heading?

cakesoccer replied:
To clarify, where exactly on the landing page will this appear? And which products will we be listing? Other OGP suite products (e.g. Postman, Go.gov.sg, Isomer) or products such as Parking.SG, RedeemSG, as well? Depending on the above and how it's supposed to look, you could say something like "Also built by OGP".

timotheeg replied:
Thanks Sarah! Apologies, I should have given you more context. 

You can see how this looks by checking the video demo in the PR description. I'm copying the video link here for you:

https://user-images.githubusercontent.com/102740235/177200059-69e08d5d-e37e-4c6b-b364-d7f82d15b98a.mp4 

The purpose of the list is to advertise to Form admins other tools that could potentially help them with their workflows, just like FormSG does. So Parking.SG isn't in the list, but the following products are:
* Go
* Postman
* CheckFirst
* Redeem
* Isomer
* Pay
* SGID

mantariksh replied:
sorry I realised that we should take out Pay because they're not taking new users anymore

tohhengxing replied:
alright, should i remove it?

timotheeg replied:
Yes please :)

tohhengxing replied:
done! 

![image](https://user-images.githubusercontent.com/102740235/177984779-844dee66-de23-49bb-95c8-9a2632fa35ae.png)

cakesoccer replied:
Ahh i see ok, so needs to be short. Maybe something like "Build Your Workflow" (though that's a bit long), "Also by OGP", "Build More", "Other Tools",  wdyt @shannenho

tohhengxing replied:
hi any updates? @cakesoccer @shannenho

shannenho replied:
sflr, I think "More Tools" or "Other Tools" work for the heading

timotheeg replied:
Thanks @shannenho . Let's pick `More Tools` then. 

@tohhengxing , since there areslight change still needed to make the css linter happy, if you could do this update as well, it would be fantastic. Thank you!

tohhengxing replied:
![image](https://user-images.githubusercontent.com/102740235/179146244-5ba78a84-e30e-4c56-a7ac-f82d7d49883a.png)

timotheeg reviewed (commented):
Thanks for the contribution @tohhengxing! The code looks good and implemented the specs from the issue definition very well 👍 .

I added 2 comments. They do not really relate to the implementation, but to the specs themselves for some discussion within the team. We'll get back to you asap!

Also to note, we are just about to complete a massive migration of the FormSG frontend to React (in another 4 to 5 weeks if all goes well). If you are up for it, porting this OGP toolset links dropdown to React would be very welcome as well 😄 !

timotheeg commented on src/public/modules/core/components/toolset-dropdown.client.component.js:
```diff
@@ -0,0 +1,34 @@
+  vm.isDropdownHover = false
+  vm.isDropdownFocused = false
+  vm.isDropdownOpen = false
+  vm.labelAndLink = [
```
Should we add a link to [parkingsg](https://www.parking.sg/) as well?

timotheeg replied:
Answer from PM @mantariksh: The links here are targeted for public officers who may not be aware of other tools that could be useful for their workflows (just like FormSG is). parkingsg doesn't fit that purpose since it's for citizens, not for officers.

tohhengxing committed:
Merge branch 'opengovsg:develop' into develop

tohhengxing committed:
Update toolset-dropdown.client.component.js

removed payments since it is no longer used as per requirements

tohhengxing committed:
Merge branch 'develop' of https://github.com/tohhengxing/FormSG-link-to-toolset-feature into develop

timotheeg reviewed (approved):
The last point of discussion on heading can be updated later. Considering the PR implements the specs, let's just merge it. 

Many thanks for your contribution @tohhengxing !

tohhengxing commented:
> The last point of discussion on heading can be updated later. Considering the PR implements the specs, let's just merge it.
> 
> Many thanks for your contribution @tohhengxing !

thank you all for reviewing my tiny contribution! I am very thankful!

timotheeg commented:
Some required checks are refusing to run for some reason, I will close and reopen the PR to unlock them.

timotheeg commented on src/public/modules/core/css/navbar.css:
```diff
@@ -107,4 +107,4 @@
     height: 76px;
   }
-}
+}
```
Please restore the end of line.

tohhengxing replied:
done, apologies i work on a windows machine and i forgot to enable the stylelint extension

timotheeg commented on src/public/modules/core/css/toolset-dropdown.css:
```diff
@@ -0,0 +1,58 @@
+}
+
+.toolset_dropdown_list_anchor {
+  margin: 0px 0px 0px 0px !important;
```
Per our css linting rules, `0` does not need units
```suggestion
  margin: 0 0 0 0 !important;
```

tohhengxing replied:
done

timotheeg reviewed (commented):
Oops, apologies, now that the linter check ran, it reported some minor issues:
```
[1] src/public/modules/core/css/navbar.css
[1]  110:2  ✖  Insert "⏎"   prettier/prettier
[1] 
[1] src/public/modules/core/css/toolset-dropdown.css
[1]  28:13  ✖  Unexpected unit   length-zero-no-unit
[1]  52:12  ✖  Unexpected unit   length-zero-no-unit
[1]  52:16  ✖  Unexpected unit   length-zero-no-unit
[1]  52:[20](https://github.com/opengovsg/FormSG/runs/7351254144?check_suite_focus=true#step:6:21)  ✖  Unexpected unit   length-zero-no-unit
[1]  52:[24](https://github.com/opengovsg/FormSG/runs/7351254144?check_suite_focus=true#step:6:25)  ✖  Unexpected unit   length-zero-no-unit
[1]
```

Could you add a commit to correct them? Thanks!

timotheeg commented on src/public/modules/core/css/toolset-dropdown.css:
```diff
@@ -0,0 +1,58 @@
+}
+
+.toolset_dropdown ul li {
+  padding: 0px;
```
Same comment here:

```suggestion
  padding: 0;
```

tohhengxing replied:
done

tohhengxing committed:
change title and fix lint issues

1. changed title from gov apps -> more tools
2. linting issues was because i did not have stylelint extension installed.
3. navbar was due to CRLF

tohhengxing committed:
Merge branch 'opengovsg:develop' into develop

timotheeg commented:
Check are not running again. I close and reopen to kick them 😅
